### PR TITLE
NONPR-3316: Upgrade nrs-retrieval-frontend to Play 3.0

### PR DESCRIPTION
**Pull request by AIbrahimQA**

### CODE REVIEW

File: app/uk/gov/hmrc/nrsretrievalfrontend/config/AuthRedirects.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.nrsretrievalfrontend.config
+
+import play.api.mvc.Result
+import play.api.mvc.Results._
+import play.api.{Configuration, Environment, Mode}
+
+trait AuthRedirects {
```
Comment (DanielConnelly) [inline app/uk/gov/hmrc/nrsretrievalfrontend/config/AuthRedirects.scala:23]: so does this file not exist within the library anymore?

File: build.sbt [hunk 2]
```diff
@@ -1,59 +1,62 @@
...
-val silencerVersion = "1.7.12"
 
 lazy val root = (project in file("."))
   .settings(
     name := appName,
     organization := "uk.gov.hmrc",
     PlayKeys.playDefaultPort := 9390,
-    majorVersion := 0,
-    scalaVersion := "2.13.8",
+    majorVersion := 1,
```
Comment (DanielConnelly) [inline build.sbt:50]: are we calling this a major version bump yeh?

File: test/uk/gov/hmrc/nrsretrievalfrontend/config/AuditableSpec.scala [hunk 3]
```diff
@@ -16,27 +16,31 @@
...
+
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutor}
 
-class AuditableSpec extends UnitSpec {
+abstract class AuditableSpec extends UnitSpec {
   private val appName = "TestAppName"
   private val mockAudit: Audit = mock[Audit]
   private val auditable = new Auditable(appName, mockAudit)
+  val ec: ExecutionContextExecutor = ExecutionContext.global
+  val dataEvent: DataEvent
```
Comment (DanielConnelly) [inline test/uk/gov/hmrc/nrsretrievalfrontend/config/AuditableSpec.scala:32]: is this needed? I can't see any changes anywhere else.

Comment (AIbrahimQA) [reply test/uk/gov/hmrc/nrsretrievalfrontend/config/AuditableSpec.scala:32]: With the new upgrade the sendDataEvent from method from the Audit library now has to take in DataEvent.

File: app/uk/gov/hmrc/nrsretrievalfrontend/config/AuthRedirects.scala (hunk 1 above)
Comment (AIbrahimQA) [reply app/uk/gov/hmrc/nrsretrievalfrontend/config/AuthRedirects.scala:23]: Nope not available anymore, i stripped it out to only include the code we needed.

**Review by DanielConnelly — APPROVED** [review]

**Pull request merged**